File: robocode2-server/src/main/java/net/robocode2/model/events/HitByBulletEvent.java
package net.robocode2.model.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bot is hit by a bullet.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class HitByBulletEvent implements IEvent {

	/** Bullet that hit the bot */
	Bullet bullet;

	/** Damage dealt to the bot */
	double damage;

	/** New energy level of the bot after damage */
	double energy;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/HitByBulletEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class HitByBulletEvent(
        val bullet: BulletState,
        val damage: Double,
        val energy: Double
) : Content(type = ContentType.HIT_BY_BULLET_EVENT.type)
